java/ql/lib/semmle/code/java/dataflow/ExternalFlow.qll

```diff
@@ -627,9 +627,13 @@ private string paramsStringPart(Callable c, int i) {
   i = 2 * c.getNumberOfParameters() and result = ")"
 }
 
-private string paramsString(Callable c) {
-  result = concat(int i | | paramsStringPart(c, i) order by i)
-}
+/**
+ * Gets a parenthesized string containing all parameter types of this callable, separated by a comma.
+ *
+ * Returns the empty string if the callable has no parameters.
+ * Parameter types are represented by their type erasure.
+ */
+string paramsString(Callable c) { result = concat(int i | | paramsStringPart(c, i) order by i) }
```

Comment:
Hmm, this PR has now acquired a semantic merge-conflict with https://github.com/github/codeql/pull/6637. @alexet what was the performance like with/without that piece of manual magic you inserted?

Reply:
Not awful, this predicate on my machine takes around 20s on the DB I was using (hadoop?). The problem is the just volume of strings that are computed in the concat as string pool encodes are a bit expensive. If we are computing this in full then we have to pay the cost so I wouldn't worry too much. It was an attempt at a fix for a 10% regression that only saved 1% so I kept it but I don't think it is worth keeping.

Reply:
Thanks for the info @alexet - I've reverted the commit for now as we need the existing `paramString` impl here and in another branch I'm working on.